### DISCUSSION TITLE
best practice in Nuxt3 to assets images

### DISCUSSION THREAD

**Discussion by zupcode** [nuxt/nuxt#19534] [Questions] [answered]
Hello,

I was wondering what is the best way to call an image in the template, not dynamicly.
I know the different ways to do but I'd like to know if there is one more efficient than others...

1 -
`<img src="~/assets/images/imgA.jpg" />`

2 -
```
<template>
<img :src="imagePath" />
...
<script setup>
import imgA from '~/assets/images/imgA.jpg'
const imagePath = imgA
```

3 -
```
<template>
<img :src="imageList.imgA" />
...
<script setup>
const imageList = useImageList()
...........
/*states.ts*/
import imgA from '~/assets/images/imgA.jpg'
import imgB from '~/assets/images/imgB.jpg'
import imgC from '~/assets/images/imgC.jpg'
...
export const useImageList = () => useState('imageList', () => ({
  imgA: imgA,
  imgB: imgB,
  imgC: imgC,
  ...
}))
```

**Comment by TheAlexLichter** [comment, marked as answer]
IMO the best is to leave the source in the template. I don't think any different way has performance implications, so option 1 should be good to go.

**Comment by zupcode** [comment]
thanks for this answer !
